source/fuzz/transformation_add_relaxed_decoration.cpp

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/transformation_add_relaxed_decoration.h"
+
+#include "source/fuzz/fuzzer_util.h"
+
+namespace spvtools {
+namespace fuzz {
+
+TransformationAddRelaxedDecoration::
+TransformationAddRelaxedDecoration(
+    const spvtools::fuzz::protobufs::
+    TransformationAddRelaxedDecoration& message)
+    : message_(message) {}
+
+TransformationAddRelaxedDecoration::
+TransformationAddRelaxedDecoration(uint32_t result_id) {
+  message_.set_result_id(result_id);
+}
+
+bool TransformationAddRelaxedDecoration::IsApplicable(
+    opt::IRContext* ir_context, const TransformationContext& transformation_context) const {
+  // |message_.result_id| must be the id of an instruction.
+  auto instr = ir_context->get_def_use_mgr()->GetDef(message_.result_id());
+  if (!instr) {
+    return false;
+  }
+  opt::BasicBlock* cur_block =
+      ir_context->get_instr_block(instr);
+  // current block must be a dead block
```

Comment:
Please start comments with a capital letter and end with a full stop (period). Prefer proper sentences. E.g. Prefer "The id must be..." vs. "Id must be...". Referring to variables is fine though: e.g. "|id| must be..." (and no capital letter is needed). 

You have done this fine but also notice: comments inside functions use the style:

"Add foo to bar."

but documentation comments before a function declaration always use the style:

"Adds..."
"Returns..."
"Transforms...", etc., 

at least for the first sentence.